pytradfri/api/aiocoap_api.py

```diff
@@ -0,0 +1,162 @@
+"""Coap implementation using aiocoap."""
+import asyncio
+import json
+import logging
+import sys
+
+sys.path.insert(0, '/usr/src/build/tinydtls/cython')  # noqa
+
+import aiocoap
+from aiocoap import Message, Context
+from aiocoap.error import RequestTimedOut
+from aiocoap.numbers.codes import Code
+from aiocoap.transports.tinydtls import TransportEndpointTinyDTLS
+
+from ..error import ClientError, ServerError, RequestTimeout
+from ..command import Command
+
+_LOGGER = logging.getLogger(__name__)
+
+aiocoap.numbers.constants.MAX_RETRANSMIT = 10
+
+
+class PatchedDTLSSecurityStore:
+    """Patched DTLS store in lieu of impl."""
+
+    SECRET_PSK = None
+
+    def _get_psk(self, host, port):
+        return b"Client_identity", PatchedDTLSSecurityStore.SECRET_PSK
+
+
+aiocoap.transports.tinydtls.DTLSSecurityStore = PatchedDTLSSecurityStore
+
+
+def _patched_datagram_received(self, data, addr):
+    self._dtls_socket.handleMessage(self._connection, data, 0)
+
+
+aiocoap.transports.tinydtls.DTLSClientConnection.datagram_received = \
+    _patched_datagram_received
+
+
+@asyncio.coroutine
+def api_factory(host, security_code):
+    """Generate a request method."""
+
+    security_code = security_code.encode('utf-8')
+
+    PatchedDTLSSecurityStore.SECRET_PSK = security_code
+
+    @asyncio.coroutine
+    def _get_protocol():
+        protocol = yield from Context.create_client_context()
+        for endpoint in protocol.transport_endpoints:
```

Comment:
Agree with @MartinHjelmare, please add a comment what this code does. Now it just looks to me like code that does nothing.